doc/design/audit-trail-mvp.md

```diff
@@ -0,0 +1,152 @@
+# Audit Trail MVP
+
+NIH has required that DANDI Archive generate audit logs while leaving undefined
+what that means. We are choosing to fulfill the requirement with a feature that
+indicates to Dandiset owners and admins the sequence of significant actions
+taken to modify Dandisets, along with identification of who performed each
+action.
+
+The major motivation is to inform owners about how their Dandisets have changed
+when a given owner did not make the change and wishes to know who to speak to
+about those changes. In essence, it is a tool to support effective
+collaboration.
+
+## Requirements
+
+1. **Persistent tracking of significant actions.** Whenever a *significant
+   action* (see list [below](#significant-actions)) occurs successfully, the archive will generate a
+   persistent record of that action, including the following data:
+    - The date and time
+    - The authorized user who triggered the action
+    - The ID of the Dandiset on which the action occurs
+    - Event-specific data about the action itself
+
+2. **Display of audit log (to owners and admins only).** The Dandiset landing
+   page will display a rendered (i.e. human-readable) version of the log event
+   data (as long as the viewing user is an owner of the Dandiset, or an admin).
+
+   A possible example of such a display follows (the final rendered form may
+   differ from this depending on best practices, etc.):
+    
+    ```
+    YYYYMMDDTHHMMSS.mmmm: X added asset at path (/full/path) checksum
+    YYYYMMDDTHHMMSS.mmmm: X modified asset at path (/full/path) checksum
+    YYYYMMDDTHHMMSS.mmmm: X deleted asset at path (/full/path) checksum
+    YYYYMMDDTHHMMSS.mmmm: X modified dandiset metadata
+    ```
```

Comment:
Would this be one row per file? That could generate a lot of row if a dandiset has many files. I don't think this is necessarily a bad thing, but if we do it this way we should plan for this, e.g. potentially paginating the rendered log

Reply:
Pagination will definitely be part of the display. This snippet is meant to show what the log will be like (independent of how we will display it).